Support ZLO device types to support newer Zigbee devices in ZHA

jurriaan opened the pull request:
## Description:

Support the device types that are added in https://github.com/zigpy/zigpy/pull/176
so newer Zigbee (3.0) devices can be supported.

Tested with my innr RB 285 C bulb.

This PR is waiting for an official zigpy release (cc @Adminiuga)

See https://github.com/zigpy/zigpy/pull/176

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. *Can be verified after release of zigpy*
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html

jurriaan committed:
Support ZLO device types

Support the device types that are added in https://github.com/zigpy/zigpy/pull/176
so newer Zigbee devices can be supported.

homeassistant commented:
Hi @jurriaan, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#24429).

Once you do that we will be able to review and accept this pull request.

Thanks!

dmulcahey reviewed (changes requested):
Please remove the mappings to Binary Sensor

dmulcahey commented on homeassistant/components/zha/core/registries.py:
```diff
@@ -115,7 +115,17 @@ def get_deconz_radio():
         zha.DeviceType.DIMMABLE_LIGHT: LIGHT,
-        zha.DeviceType.COLOR_DIMMABLE_LIGHT: LIGHT
+        zha.DeviceType.COLOR_DIMMABLE_LIGHT: LIGHT,
+        zha.DeviceType.ON_OFF_LIGHT_SWITCH: BINARY_SENSOR,
```
Remove this as we no longer use binary sensor for remotes.

jurriaan replied:
Fixed these instance by matching them to SWITCH and LIGHTs since they are target devices according to the Zigbee documentation.

dmulcahey commented on homeassistant/components/zha/core/registries.py:
```diff
@@ -115,7 +115,17 @@ def get_deconz_radio():
-        zha.DeviceType.COLOR_DIMMABLE_LIGHT: LIGHT
+        zha.DeviceType.COLOR_DIMMABLE_LIGHT: LIGHT,
+        zha.DeviceType.ON_OFF_LIGHT_SWITCH: BINARY_SENSOR,
+        zha.DeviceType.DIMMER_SWITCH: BINARY_SENSOR,
```
Remove this

dmulcahey commented on homeassistant/components/zha/core/registries.py:
```diff
@@ -115,7 +115,17 @@ def get_deconz_radio():
+        zha.DeviceType.COLOR_DIMMABLE_LIGHT: LIGHT,
+        zha.DeviceType.ON_OFF_LIGHT_SWITCH: BINARY_SENSOR,
+        zha.DeviceType.DIMMER_SWITCH: BINARY_SENSOR,
+        zha.DeviceType.COLOR_DIMMER_SWITCH: BINARY_SENSOR,
```
Remove this

dmulcahey commented on homeassistant/components/zha/core/registries.py:
```diff
@@ -115,7 +115,17 @@ def get_deconz_radio():
+        zha.DeviceType.DIMMABLE_PLUG_IN_UNIT: LIGHT,
+        zha.DeviceType.COLOR_TEMPERATURE_LIGHT: LIGHT,
+        zha.DeviceType.EXTENDED_COLOR_LIGHT: LIGHT,
+        zha.DeviceType.ON_OFF_SENSOR: BINARY_SENSOR
```
Remove this

jurriaan replied:
What should sensors map to? Or how does this work for sensors?

dmulcahey replied:
Remote devices don’t need the mappings. They emit events.

jurriaan replied:
Ah sorry, makes total sense. Will fix this!

dmulcahey replied:
no need to apologize. We just made this change :)

jurriaan committed:
Remove BINARY_SENSOR mappings

jurriaan committed:
Add back ON_OFF_LIGHT_SWITCH, DIMMER_SWITCH and COLOR_DIMMABLE_LIGHT

Since they are target devices I've added them as switch and lights,
which matches the Zigbee documentation.

dmulcahey commented:
can merge once builds pass

Adminiuga commented:
Um, we have to push new zigpy version to PyPi for the build to pass.

jurriaan commented:
@Adminiuga correct, mentioned that in this PR description. Is it possible for you to push a new version?

Adminiuga commented:
@jurriaan just pushed new version to PyPi. Update the manifest.json and run `script/gen_requirements_all.py`

@dmulcahey is it ok to bump version for zigpy in this PR you would you like a new one?

jurriaan committed:
Upgrade to zigpy-homeassistant v0.5.0

To be able to use the new DeviceTypes

dmulcahey commented:
Same is fine by me...

Adminiuga commented:
aprs device tracker breaks this build...

Adminiuga reviewed (approved):
Approved for merging.

Adminiuga commented:
@jurriaan do you want to try rebasing it to prior https://github.com/home-assistant/home-assistant/commit/8fcfcc40fccb606e7dfe4b79c41ec3dcb408e6c5#diff-34298aba7fdfcffa995d0a1fd0b2a068 commit? To exclude `homeassistant.components.aprs`, as it seems device tracker is breaking the builds.

jurriaan commented:
@Adminiuga Rebased and green
